client/src/pages/Calendar.js

```diff
@@ -0,0 +1,104 @@
+import React, { useEffect, useState } from "react";
+import Alert from "react-bootstrap/Alert";
+
+const Calendar = () => {
+	const [bookingsData, setBookingsData] = useState([]);
+	const [error, setError] = useState(null);
+	const [isLoading, setIsLoading] = useState(false);
+
+	useEffect(() => {
+		setIsLoading(true);
+		fetch("/api/bookings?filter=2weeks")
+			.then((response) => {
+				if (!response.ok) {
+					throw Error(
+						"There was an error getting the calendar data, Please try again"
+					);
+				}
+				return response.json();
+			})
+			.then((data) => {
+				setBookingsData(data);
+				setError(null);
+			})
+			.catch((err) => {
+				setError(err.message);
+			})
+			.finally(() => {
+				setIsLoading(false);
+			});
+	}, []);
+
+	if (isLoading) {
+		return (
+			<div className="d-flex justify-content-center">
+				<div className="spinner-grow">
+					<span className="visually-hidden">Loading...</span>
+				</div>
+			</div>
+		);
+	}
+	if (error != null) {
+		return <Alert variant="danger">{error}</Alert>;
+	}
+
+	const pickUpDates = [];
```

Comment:
```suggestion
        // Each element has the structure { date: Date, clients: [] }
	const pickUpDates = [];
```
Maybe we can leave a comment to quickly show what the structure in `pickUpDate` is